File: 백준/Silver/1789. 수들의 합/수들의 합.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit
n=int(input())
sum=0
cnt=0
for i in range(1, n+1):
    sum+=i
    cnt+=1
    if sum>n:
        cnt-=1
        break
if cnt<0:
    cnt=0
print(cnt)

File: 백준/Silver/1789. 수들의 합/수들의 합.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	var sum, cnt int

	fmt.Fscanf(reader, "%d\n", &n)
	for i := 1; i <= n; i++ {
		sum += i
		cnt++
		if sum > n {
			cnt -= 1
			break
		}
	}
	if cnt <= 0 {
		fmt.Println(0)
	} else {
		fmt.Println(cnt)
	}
}
